storage/integration_test/src/integration_test.cc

```diff
@@ -386,6 +398,112 @@ const std::string kSimpleTestFile =
     "pariatur. Excepteur sint occaecat cupidatat non proident, sunt in "
     "culpa qui officia deserunt mollit anim id est laborum.";
 
+// Only test retries on desktop since Android and iOS don't have an option
+// to retry file-not-found errors and just pass-through to native
+// implementations.
+#if FIREBASE_PLATFORM_DESKTOP
+// Set a test-only flag to enable retrying 404s.
+TEST_F(FirebaseStorageTest, TestReadBytesWithMaxRetryDuration) {
+  // Enable retrying of file-not-found errors for testing.
+  bool old_value = firebase::storage::internal::g_retry_file_not_found__for_testing;
+  firebase::storage::internal::g_retry_file_not_found__for_testing = true;
+
+  int shorter_duration_sec = 2;
+  int longer_duration_sec = 6;
+  SignIn();
+
+  // Attempt to read a non-existent file, and verify that GetBytes fails.
+  {
+    LogDebug("Call GetBytes on a nonexistent ref.");
+    firebase::storage::StorageReference ref = CreateFolder().Child("File1.txt");
+    LogDebug("Storage URL: gs://%s%s", ref.bucket().c_str(),
+             ref.full_path().c_str());
+    cleanup_files_.push_back(ref);
```

Comment:
You only need to add ref to cleanup_files if a file was uploaded into that ref.

Although I think it might be best to use the same ref for the first test (to ensure the file starts out missing) and the second test (to ensure that it retries on the file that we previously confirmed is missing).